modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -405,3 +421,58 @@ ModularGapConductanceConstraint::computeQpResidual(Moose::MortarType
   mooseError("We should never get here. We should have errored in the constructor.");
 #endif
 }
+
+void
+ModularGapConductanceConstraint::deduceGeometryParameters()
+{
+  Point position;
+  Real area = 0.0;
+  const auto my_pid = processor_id();
+
+  // build side element list as (element, side, id) tuples
+  const auto bnd = _mesh.buildActiveSideList();
+
+  std::unique_ptr<const Elem> side_ptr;
+  for (auto [elem_id, side, id] : bnd)
+    if (id == _primary_id)
+    {
+      const auto * elem = _mesh.elemPtr(elem_id);
+      if (elem->processor_id() != my_pid)
+        continue;
+
+      // update side_ptr
+      elem->side_ptr(side_ptr, side);
+
+      // area of the (linearized) side
+      const auto side_area = side_ptr->volume();
+
+      // position of the side
+      const auto side_position = side_ptr->true_centroid();
+
+      // sum up
+      position += side_position * side_area;
+      area += side_area;
+    }
+
+  // parallel communication
+  _communicator.sum(position);
+  _communicator.sum(area);
+
+  // set axis
+  if (area == 0.0)
+  {
+    if (_gap_geometry_type == GapGeometry::CYLINDER)
+      paramError("gap_geometry_type",
+                 "Unable to decuce cylinder axis automatically, please specify "
+                 "'cylinder_axis_point_1' and 'cylinder_axis_point_2'.");
+    else if (_gap_geometry_type == GapGeometry::SPHERE)
+      paramError("gap_geometry_type",
+                 "Unable to decuce sphere origin automatically, please specify "
+                 "'sphere_origin'.");
+    else
+      mooseError("Internal error.");
```

Comment:
Can provide more details as to what generates this error.

Reply:
The method should only be called for the cylindrical and spherical cases in the first place, so we should never get to this internal error. Only a developer could trigger this while changing the code. This is defensive programming to alert a developer and have them think about what they just did.